### ISSUE TITLE
If there is no tree connect message the authentication is ignored by ntlmrelayx.py (PetitPotam)

### ISSUE THREAD

**Issue by micahvandeusen** [SecureAuthCorp/impacket#1138]
### Configuration  
impacket version:  Impacket 0.9.23
Python version:  Python 3.9.2
Target OS: 
Connection from Windows 6.2 Build 9200 x64
Kali 2021.2
Target Windows 10.0 Build 14393 x64

### Debug Output With Command String  
python3 examples/ntlmrelayx.py -t http://AD CS Host/certsrv/certfnsh.asp -debug -smb2support

### PCAP  
This was on a client network so I don't have this information.

### Additional context  
When using the latest version of impacket with PetitPotam relaying from SMB to HTTP, ntlmrelayx does not even show that a connection happened. Looking at the packet capture it appears that the machine connecting is not sending a tree connect message after the original authentication. Using an older version of impacket before the changes mentioned here https://www.secureauth.com/blog/what-is-old-is-new-again-the-relay-attack/ I was able to get it to work. It would be nice to have the ability to choose to use the multi relay.

**Comment by 0xdeaddood** [comment]
Hi @micahvandeusen! 

I think there is something else here. After a successfully authentication, the client sends a TreeConnect request against the IPC$ share. I've tested PetitPotam with the latest impacket version, relaying from SMB to HTTP, and it works great -> I can execute the AD CS attack implementation (#1101 by @ExAndroidDev). 

I need more information to analyze this case. Could you share the debug output? If you can reproduce the issue on a different network, could you send me a PCAP capture?

**Comment by micahvandeusen** [comment]
There is no debug output at all from impacket and no error messages.

In a lab environment it works fine. However, it has been in two client environments now that this has been an issue. I'm not sure what specifically is causing this and is different.

**Comment by micahvandeusen** [comment]
Found it. Setting the following allows me to simulate this in a lab:
MACHINE\System\CurrentControlSet\Services\LanManServer\Parameters\AutoDisconnect=4,15
MACHINE\System\CurrentControlSet\Services\LanManServer\Parameters\EnableForcedLogOff=4,1
MACHINE\System\CurrentControlSet\Services\LanManServer\Parameters\EnableSecuritySignature=4,1
MACHINE\System\CurrentControlSet\Services\LanManServer\Parameters\NullSessionPipes=7,,netlogon,samr,lsarpc
MACHINE\System\CurrentControlSet\Services\LanManServer\Parameters\RequireSecuritySignature=4,1
MACHINE\System\CurrentControlSet\Services\LanManServer\Parameters\RestrictNullSessAccess=4,1
MACHINE\System\CurrentControlSet\Services\LanmanWorkstation\Parameters\EnablePlainTextPassword=4,0
MACHINE\System\CurrentControlSet\Services\LanmanWorkstation\Parameters\EnableSecuritySignature=4,1
MACHINE\System\CurrentControlSet\Services\LanmanWorkstation\Parameters\RequireSecuritySignature=4,1

**Comment by micahvandeusen** [comment]
This seems to affect any SMB connections where the client enforces signing.
The error on the client side is "Invalid Signature."

**Comment by 0xdeaddood** [comment]
Hi @micahvandeusen!

We've verified the issue, thanks for reporting it. If the client enforces signing, the connection finishes before we can relay it. 
We're thinking about the best way to fix it, we'll probably share here some alternatives to discuss. Meanwhile, as you mentioned, using version 0.9.20 or earlier solves the problem.

**Comment by 0xdeaddood** [comment]
Hi there! 

FYI, I just added a new option `--no-multirelay` to disable multi-relaying capabilities (https://github.com/SecureAuthCorp/impacket/pull/1297). Useful in some scenarios like this one where the client requires SMB signing. The following command should works:

`ntlmrelayx.py -t http://HOST/certsrv/certfnsh.asp -debug -smb2support --no-multirelay`

Closing. Reopen if you have any questions or if you find something not working.

Thanks!
